src/test/java/seedu/address/storage/JsonAdaptedEventTest.java

```diff
@@ -0,0 +1,172 @@
+    public void toModelType_invalidRecurFrequency_throwsIllegalValueException() {
+        JsonAdaptedEvent event = new JsonAdaptedEvent(
+                VALID_EVENT_DESCRIPTION, VALID_DATE, VALID_TIME, VALID_DURATION, INVALID_RECURFREQUENCY);
+        String expectedMessage = RecurFrequency.INVALID_RECUR_FREQUENCY_MESSAGE;
+        assertThrows(IllegalValueException.class, expectedMessage, event::toModelType);
+    }
+
+    @Test
+    public void toModelType_nullRecurFrequency_throwsIllegalValueException() {
+        JsonAdaptedEvent event = new JsonAdaptedEvent(
+                VALID_EVENT_DESCRIPTION, VALID_DATE, VALID_TIME, VALID_DURATION, null);
+        String expectedMessage = String.format(MISSING_FIELD_MESSAGE_FORMAT, RecurFrequency.class.getSimpleName());
+        assertThrows(IllegalValueException.class, expectedMessage, event::toModelType);
+    }
+
+    @Test
+    public void toModelType_updateDaily_success() throws Exception {
+        JsonAdaptedEvent event = new JsonAdaptedEvent(
+                VALID_EVENT_DESCRIPTION, PAST_RESET_DATE_ODD, VALID_TIME, VALID_DURATION, RECURFREQUENCY_DAILY);
+        LocalDate expectedDate = LocalDate.now();
+        assertEquals(expectedDate, event.toModelType().getDate());
+    }
+
+    @Test
+    public void toModelType_updateWeekly_success() throws Exception {
+        JsonAdaptedEvent event = new JsonAdaptedEvent(
+                VALID_EVENT_DESCRIPTION, PAST_RESET_DATE_ODD, VALID_TIME, VALID_DURATION, RECURFREQUENCY_WEEKLY);
+        LocalDate expectedDate = LocalDate.now().with(next(PAST_RESET_DATE_ODD_DAYOFWEEK));
+        assertEquals(expectedDate, event.toModelType().getDate());
+    }
+
+    @Test
+    public void toModelType_updateOddWeekBiweekly_success() throws Exception {
+        JsonAdaptedEvent event = new JsonAdaptedEvent(
+                VALID_EVENT_DESCRIPTION, PAST_RESET_DATE_ODD, VALID_TIME, VALID_DURATION, RECURFREQUENCY_BIWEEKLY);
+        LocalDate today = LocalDate.now();
+        LocalDate expectedDate = today.with(next(PAST_RESET_DATE_ODD_DAYOFWEEK));
+        if (ChronoUnit.DAYS.between(PAST_RESET_DATE_ODD_LOCALDATE, expectedDate) % 14 != 0) {
+            expectedDate = expectedDate.with(next(PAST_RESET_DATE_ODD_DAYOFWEEK));
+        }
+        assertEquals(expectedDate, event.toModelType().getDate());
+    }
+
+    @Test
+    public void toModelType_updateEvenWeekBiweekly_success() throws Exception {
+        JsonAdaptedEvent event = new JsonAdaptedEvent(
+                VALID_EVENT_DESCRIPTION, PAST_RESET_DATE_EVEN, VALID_TIME, VALID_DURATION, RECURFREQUENCY_BIWEEKLY);
+        LocalDate today = LocalDate.now();
+        LocalDate expectedDate = today.with(next(PAST_RESET_DATE_EVEN_DAYOFWEEK));
+        if (ChronoUnit.DAYS.between(PAST_RESET_DATE_EVEN_LOCALDATE, expectedDate) % 14 != 0) {
+            expectedDate = expectedDate.with(next(PAST_RESET_DATE_EVEN_DAYOFWEEK));
+        }
+        assertEquals(expectedDate, event.toModelType().getDate());
+    }
```

Comment:
This can also be combined? `toModelType_updateRecurrence_success`

Reply:
Also should this method throw an Exception when we are testing?

Reply:
Following what the `JsonAddressStorageBook` tests, I think its okay to throw Exceptions. I am not sure how else to deal with it. It doesn't quite make sense to catch the exception inside the test itself also.